src/utils/provider.ts

```diff
@@ -0,0 +1,117 @@
+import { FallbackProvider, JsonRpcProvider, WebSocketProvider } from 'ethers';
+import { FallbackProviderConfig } from 'ethers/lib.commonjs/providers/provider-fallback';
+import { isDefined } from './util';
+
+export type FallbackProviderJsonConfig = {
+  chainId: number;
+  providers: ProviderJson[];
+};
+
+export type ProviderJson = {
+  priority: number;
+  weight: number;
+  provider: string;
+  chainId?: number;
+  stallTimeout?: number;
+  maxLogsPerBatch?: number;
+};
+
+export const createProviderFromJsonConfig = (
+  config: FallbackProviderJsonConfig | ProviderJson,
+  pollingInterval?: number,
+): FallbackProvider | WebSocketProvider | JsonRpcProvider => {
+  try {
+    // Handle single provider case
+    if (!('providers' in config)) {
+      // Get RPC URL
+      const providerURL = config.provider;
+      // Ensure providerURL exists and is a string
+      if (!isDefined(providerURL)) {
+        throw new Error('provider is required for single provider configuration');
```

Comment:
Just as a reminder, lets try to make this errors look similar to https://github.com/Railgun-Community/shared-models/pull/38

Reply:
Could you provide an example? Not clear what change is being asked for